staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -169,6 +169,11 @@ func (d *DiscoveryClient) ServerGroups() (apiGroupList *metav1.APIGroupList, err
 	if err != nil && !errors.IsNotFound(err) && !errors.IsForbidden(err) {
 		return nil, err
 	}
+	// populate the hash for the legacy core group
+	d.hashesMu.Lock()
```

Comment:
I'm not clear about why this bit of code has to be located here with an additional lock/unlock, rather than below